docs/readmes/proposals/p025_Deep_packet_Inspection.md

```diff
@@ -0,0 +1,124 @@
+[https://github.com/magma/magma/issues/11134](https://github.com/magma/magma/issues/11134).
+
+## Overview
+DPI (Deep Packet Inspection) is a traffic recognition method that classifies IP traffic in real time. DPI identifies protocols, applications and application attributes. In addition to IP traffic classification, DPI engines extract protocol and application-based metadata,
+providing insight into user behavior and application usage. Examples of metadata that can be
+extracted from IP traffic include the following:
+
+Metadata category                        Example metadata
+Traffic volume 					Per user, per protocol, per application, per flow, per direction.
+						
+Service detection 				Differentiation between for example Skype audio and video calls
+
+Quality of service              Jitter, throughput, latency, roundtrip time, ramp-up time, packet loss,retransmissions
+
+## Purpose
+TIP OCN Functional requirement REQ-OCN-18 for FWA deployment asks for application detection as a minimum DPI requirement. 
+Adding DPI support on Magma will help enterprises that subscribe to FWA services get the following benefits
+1)	DPI is required to deliver real-time intelligence about traffic to create the most effective solution.
+2)	With DPI-enabled FWA Deployments, operators have the ability to tailor policies and adjust the traffic shaper based on time, package, or applications.
+3)	DPI will be applied to improve network efficiency, but ultimately it will allow carriers to deliver tailored Edge / Application services that increase customer satisfaction, create differentiation, and provide revenue growth.
+
+4) Policy and charging control: Provides policy control and charging software vendors with the capability to define bandwidth guarantees,
+priorities and limits, offer fine-grained QoS for an additional fee and deliver real-time charging and billing support. It can offers a high detection rate (> 95 %) and accurate application identification for policy and billing purposes.
+
+5) Enterprises can leverage DPI to block or throttle access to risky or unauthorized applications, block policy-violating usage patterns or prevent unauthorized data access within corporate-approved applications, stop data exfiltration attempts by external attackers or potential data leaks caused by both malicious and negligent insiders.
+
+Wavelabs as a VAR (Value Added Reseller) for magma 5G SA core received RFPs from FWA service providers. DPI is a feature asked by all the service providers. 
+
+## Gap (Design) Analysis
+To provide DPI support on magma  OVS and Pipelined will need the following changes;
+1.  Provide the new Opaque DPI Interface that integrates with the open source DPI plugin. (dynamically any 3rd party / external / customer DPI engine can be hooked with OVS seamlessly)
+2.  Implement the sample plugin will demonstrate the DPI public API that was given for DPI-enabled-OVS. This sample plugin will simply write out the ethernet packet from OVS to a file.

```

Comment:
This thing looks ancient https://github.com/kspviswa/dpi-enabled-ovs, any idea if it'll work with our ovs version? Or are you intending to update it?